Now that it builds, we face our old nemesis, Dialyzer:

```
==> riak_kv (dialyzer)
riak_kv_pb_timeseries.erl:515: The variable SQL can never match since previous clauses completely covered the type {'error',_}
riak_kv_pb_timeseries.erl:525: Function convert_cover_list/2 will never be called
riak_kv_pb_timeseries.erl:545: Function assemble_ts_range/2 will never be called
riak_kv_pb_timeseries.erl:562: Function sql_to_cover/4 will never be called
riak_kv_pb_timeseries.erl:578: Function reverse_sql/1 will never be called
riak_kv_pb_timeseries.erl:588: Function derive_description/3 will never be called
riak_kv_pb_timeseries.erl:596: Function pick_operator/2 will never be called
riak_kv_pb_timeseries.erl:601: Function extract_time_boundaries/2 will never be called
riak_kv_pb_timeseries.erl:613: Function identify_quantum_field/1 will never be called
riak_kv_pb_timeseries.erl:618: Function find_hash_fn/1 will never be called
riak_kv_pb_timeseries.erl:643: The pattern {'ok', Queries} can never match the type {'error','full table scan not implemented' | 'query is already compiled'}
riak_kv_pb_timeseries.erl:658: The call riak_kv_pb_timeseries:make_tsqueryresp(Data::[[any(),...]]) will never return since it differs in the 1st argument from the success typing arguments: ({_,_,[[boolean() | binary() | [any()] | number()]]})
riak_kv_pb_timeseries.erl:810: Function make_tsqueryresp/1 has no local return
riak_kv_pb_timeseries.erl:810: The pattern {_, _, []} can never match the type [[any(),...]]
riak_kv_pb_timeseries.erl:812: The pattern {ColumnNames, ColumnTypes, JustRows} can never match the type [[any(),...]]
riak_kv_qry.erl:102: The pattern {'ok', Queries} can never match the type {'error','full table scan not implemented' | 'query is already compiled'}
riak_kv_qry.erl:110: Function maybe_await_query_results/1 will never be called
make: *** [dialyzer-run] Error 1
```

--- a/src/riak_kv_qry_compiler.erl
+++ b/src/riak_kv_qry_compiler.erl
@@ -39,15 +39,17 @@
     {ok, [?SQL_SELECT{}]} | {error, any()}.
 compile(#ddl_v1{}, ?SQL_SELECT{is_executable = true}, _MaxSubQueries) ->
     {error, 'query is already compiled'};
-compile(#ddl_v1{}, ?SQL_SELECT{'SELECT' = #riak_sel_clause_v1{ clause = [] } }, _MaxSubQueries) ->
-    {error, 'full table scan not implemented'};
 compile(#ddl_v1{} = DDL,
-        ?SQL_SELECT{is_executable = false, type = sql} = Q, MaxSubQueries) ->
-    case compile_select_clause(DDL, Q) of
-        {ok, S} ->
-            compile_where_clause(DDL, Q?SQL_SELECT{'SELECT' = S}, MaxSubQueries);
-        {error, _} = Error ->
-            Error
+        ?SQL_SELECT{is_executable = false, 'SELECT' = Sel} = Q, MaxSubQueries) ->
+    if Sel#riak_sel_clause_v1.clause == [] ->
+            {error, 'full table scan not implemented'};
+        el/=se ->
+            case compile_select_clause(DDL, Q) of
+                {ok, S} ->
+                    compile_where_clause(DDL, Q?SQL_SELECT{'SELECT' = S}, MaxSubQueries);
+                {error, _} = Error ->
+                    Error
+            end
     end.
 
 %% adding the local key here is a bodge

